### PR TITLE
Fix volume snapshot of encrypted NFS/StorPool volume

### PR DESCRIPTION
**Pull request by slavkap**
### Description

Fix volume snapshot of encrypted NFS volume.

```
2024-04-03 15:39:03,835 ERROR [kvm.storage.KVMStorageProcessor] (agentRequest-Handler-1:null) (logid:7f88a6bf) Failed take snapshot for volume [volumeTO[uuid=2cc4e5b3-37e1-437f-928e-13c00b73
caf7|path=2cc4e5b3-37e1-437f-928e-13c00b73caf7|datastore=PrimaryDataStoreTO[uuid=cc7256f6-831e-3c38-86ec-dc785413ce21|name=nfs|id=3|pooltype=NetworkFilesystem]]], in VM [i-2-21-VM], due to [
Failed to convert volumeTO[uuid=2cc4e5b3-37e1-437f-928e-13c00b73caf7|path=2cc4e5b3-37e1-437f-928e-13c00b73caf7|datastore=PrimaryDataStoreTO[uuid=cc7256f6-831e-3c38-86ec-dc785413ce21|name=nfs|id=3|pooltype=NetworkFilesystem]] snapshot of volume [/mnt/cc7256f6-831e-3c38-86ec-dc785413ce21/2cc4e5b3-37e1-437f-928e-13c00b73caf7] to [/mnt/cc7256f6-831e-3c38-86ec-dc785413ce21/snapshots
/e620de51-d0ec-4600-98b6-c1302b10b125] due to [qemu-img: Could not open 'driver=qcow2,file.filename=/mnt/cc7256f6-831e-3c38-86ec-dc785413ce21/2cc4e5b3-37e1-437f-928e-13c00b73caf7': Parameter
 'encrypt.key-secret' is required for cipher].].
```
 
Fix backup snapshot to secondary of encrypted StorPool volume

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [X] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Manually tested with StorPool/NFS
executed some StorPool smoke tests

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan test

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java [hunk 1]
```diff
@@ -1936,26 +1938,38 @@ protected void manuallyDeleteUnusedSnapshotFile(boolean isLibvirtSupportingFlagD
...
+            String srcKeyName = "sec0";
+            List<QemuObject> qemuObjects = new ArrayList<>();
+            Map<String, String> options = new HashMap<>();
+            QemuImageOptions qemuImageOpts = new QemuImageOptions(baseFile.getPath());
+            if (srcKey.isSet()) {
+                qemuObjects.add(QemuObject.prepareSecretForQemuImg(baseFile.getFormat(),
+                        EncryptFormat.LUKS, srcKey.toString(), srcKeyName, options));
+                qemuImageOpts = new QemuImageOptions(baseFile.getFormat(), baseFile.getPath(), srcKeyName);
+            }
+            QemuImgFile srcFile = new QemuImgFile(baseFile.getPath());
```
[the remark this answers is not in the document]
Comment (slavkap) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java]: thanks, @DaanHoogland for the review!
The requested change has been addressed

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
Rekicked GH Actions

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Comment by DaanHoogland** [comment]
not sure why, but the GHA keep failing

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Pull request merged**